### PR TITLE
Prepare for ORM split

### PR DESCRIPTION
**Pull request by markstory**
I think this is what is required to do an ORM subtree split. I've not done many of the splits though.

### CODE REVIEW

**Comment by lorenzo** [comment]
Thanks for starting this @markstory

File: src/ORM/composer.json [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+	"require": {
+		"cakephp/collection": "dev-master",
+		"cakephp/core": "dev-master",
+		"cakephp/datasource": "dev-master",
+		"cakephp/database": "dev-master",
+		"cakephp/event": "dev-master",
+		"cakephp/i18n": "dev-master",
+		"cakephp/utility": "dev-master",
+		"cakephp/validation": "dev-master"
+	},
```
Comment (lorenzo) [inline src/ORM/composer.json]: We should at least `suggest` the cache package. `i18n` could also be suggested instead of required.

Comment (ADmad) [inline src/ORM/composer.json]: Yup `i18n` should be moved to suggested.

**Comment by markstory** [comment]
Updated composer.json file.

File: src/ORM/README.md [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+
+The CakePHP ORM provides a powerful and flexible way to work with relational
+databases. Using a datamapper pattern the ORM allows you to manipulate data as
+entities allowing you to create expressive domain layers in your applications.
+
+## Connecting to the Database
+
+The CakePHP ORM is compatible with:
+
+* MySQL 5.1+
```
Comment (ionas) [inline src/ORM/README.md]: Just in general, do you really want to "officially" support (e.g. warrant) of such an old version of MySQL? Why not 5.5+? ... that's from 2010?

Comment (markstory) [inline src/ORM/README.md]: The ORM doesn't generate and SQL that won't work on 5.1 so we can 'support' it until it is a problem.

**Pull request merged**